tbdex/orderstatus/orderstatus.go

```diff
@@ -0,0 +1,169 @@
+package orderstatus
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/TBD54566975/tbdex-go/tbdex"
+	"github.com/gowebpki/jcs"
+	"github.com/tbd54566975/web5-go/dids/did"
+	"go.jetpack.io/typeid"
+)
+
+// Kind identifies this message kind
+const Kind = "orderstatus"
+
+// OrderStatus represents an order status message within the exchange.
+type OrderStatus struct {
+	MessageMetadata *tbdex.MessageMetadata `json:"metadata,omitempty"`
+	Data            *Data                  `json:"data,omitempty"`
+	Signature       string                 `json:"signature,omitempty"`
+}
+
+// Data encapsulates the data content of an order status.
+type Data struct {
+	OrderStatus string `json:"orderStatus,omitempty"`
+}
+
+// Digest computes a hash of the message
+func (os OrderStatus) Digest() ([]byte, error) {
+	payload := map[string]any{"metadata": os.MessageMetadata, "data": os.Data}
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return nil, fmt.Errorf("failed to JSON marshal order status: %w", err)
+	}
+
+	canonicalized, err := jcs.Transform(payloadBytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to canonicalize order status: %w", err)
+	}
+
+	hasher := sha256.New()
+	_, err = hasher.Write(canonicalized)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compute digest: %w", err)
+
+	}
+
+	return hasher.Sum(nil), nil
+}
```

Comment:
yo @phoebe-lew any reason behind not using `tbdex.DigestJSON` like we have for other message kinds?

Reply:
oh! we're not using it in `offering`. will update both